packages/dht/src/identifiers.ts

```diff
@@ -28,3 +28,16 @@ export const createRandomDhtAddress = (): DhtAddress => {
     return getDhtAddressFromRaw(crypto.randomBytes(KADEMLIA_ID_LENGTH_IN_BYTES))
 }
 
+export const serializePeerDescriptorForSigning = (peerDescriptor: PeerDescriptor): Uint8Array => {
+    const undefinedBuffer = Buffer.from('undefined')
```

Comment:
If a field is undefined, we could use `new Uint8Array(0)`, similarly to StreamrClient:
https://github.com/streamr-dev/network/blob/a7c2de649b918b33c8f755cb43131d779f4bd560/packages/protocol/src/protocol/message_layer/signature.ts#L28-L29

Reply:
Then there is risk of confusion, the following will have the same serialization:
{
1: 121,
2: undefined
3: 345
}

{
1: 121,
2: 345
3: undefined
}

Reply:
this is exactly the kind of source of error I wanted to avoid by placing the signed part in a separate field and using standard protobuf serialization

Reply:
Either approach is fine IMO